### PR TITLE
Use for_each instead of count

### PR DESCRIPTION
**Pull request by Nuru**
## what

- Optionally use `for_each` instead of `count` when creating routes
- Avoid setting `transit_gateway_default_route_table_association` and `transit_gateway_default_route_table_propagation` in `aws_ec2_transit_gateway_vpc_attachment`

## why

- Avoid unnecessary changes when making necessary ones
- These were not previously configurable (always set to `false`) and configuring them is not supported when using Resource Access Manager, so setting them `false` caused Terraform to always plan changes to them. At least by leaving them unset/default we can avoid constant plan changes.

## references

- Terraform [documentation on VPC attachment](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_transit_gateway_vpc_attachment)

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -20,16 +24,33 @@ resource "aws_ec2_transit_gateway_route_table" "default" {
   tags               = module.this.tags
 }
 
+# Need to find out of VPC is in same account as Transit Gateway.
```
Comment (aknysh) [inline main.tf:27]:
```suggestion
# Need to find out if VPC is in same account as Transit Gateway.
```

**Review by aknysh — COMMENTED** [review]
one typo, otherwise LGTM

**Review by aknysh — APPROVED** [review]

**Review by htplbc — APPROVED** [review]

**Pull request merged**